src/core/mac/mac.cpp

```diff
@@ -214,10 +214,76 @@ bool Mac::IsEnergyScanInProgress(void)
 void Mac::StartEnergyScan(void)
 {
     mState = kStateEnergyScan;
-    mEnergyScanCurrentMaxRssi = kInvalidRssiValue;
-    mMacTimer.Start(mScanDuration);
-    mEnergyScanSampleRssiTask.Post();
-    NextOperation();
+
+    if (!(otPlatRadioGetCaps(NULL) & kRadioCapsEnergyScan))
```

Comment:
If this merges before #357, then I will have to update it to replace all these NULLs with `mNetif.GetInstance()`. Otherwise, I will have to update #357 to do the replacement.